### ISSUE TITLE
Add custom button without having to create a blade file

### ISSUE THREAD

**Issue by tabacitu** [Laravel-Backpack/community-forum#149]
TLDR:

**The problem:** for each button you create, even if it's a quick/simple one, you have to create a blade file;

**The solution:** a method that uses a general-purpose blade file, so you don't have to create any blade files;

----

Most of my button views looks like this anyway:
```php
@if ($crud->hasAccess('create'))
	<a href="{{ url($crud->route.'/create') }}" class="btn btn-primary" data-style="zoom-in"><span class="ladda-label"><i class="la la-plus"></i> {{ trans('backpack::crud.add') }} {{ $crud->entity_name }}</span></a>
@endif
```

So I imagine we can create a blade file that takes in all those variables, and outputs a simple button. So something like this pseudocode:

```php
@php
    $access = $button['access'] ?? true;
    $url = $button['url'] ? url('$button['access']') : '';
    $label = $button['label'] ?? '';
    $icon = $button['icon'] ?? 'la la-question';
@endphp

@if ($crud->hasAccess($access))
	<a href="{{ $url }}" class="btn btn-primary" data-style="zoom-in"><span class="ladda-label"><i class="{{ $icon }}"></i> {{ $label }}</span></a>
@endif
```

Which would be loaded from the CrudController using something like:
```php
$this->crud->addQuickButton();
```

Then again, we might need to do three separate methods, because depending on the stack... they could need three different blade templates, so we might end up with:

```php
// normal syntax (bleah)
$this->crud->addLineButton($name, $icon, $label, $position, $url, $access);
$this->crud->addTopButton($name, $icon, $label, $position, $url, $access);
$this->crud->addBulkButton($name, $icon, $label, $position, $url, $access);

// fluent syntax (better, but needs more work)
CRUD::lineButton('open')->icon('la la-external-link-alt')->label('Open')->url(function($entry) {
        return $entry->link;
    });
CRUD::topButton('create')->icon('la la-plus')->label('Create')->url($this->crud->route.'/create');
CRUD::bulkButton('clone'); // TBD
```

Alternatively, it might be possible to do all of the above... without considerably changing anything... or adding new syntax... by making the "button template view" the default button view, and load it in case the dev hasn't specified a custom view.

The whole idea needs to be polished. But I just wanted to get it out on my head, into an issue, so we can talk about it and do it. Bottom line is... creating a button, or at least the most common button types... should probably be a one-liner.

**Comment by pxpm** [comment]
Hey @tabacitu 

I think the cherry on top of the cake would be:
```php
CrudButton::line('open')->icon('la-la-icon')
CrudButton::top('create')
    ->icon('la-la-icon')
    ->label('Create')
    ->access(fn() => backpack_user()->hasRole('admin'))
```
This would be possible after https://github.com/Laravel-Backpack/CRUD/pull/4819 gets merged I think.

**Comment by pxpm** [comment]
https://backpackforlaravel.com/docs/6.x/crud-buttons#creating-a-quick-button-1 🔥
